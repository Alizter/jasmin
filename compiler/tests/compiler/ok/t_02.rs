fn foo(x: stack! (b64),  y: stack! (b64), z: reg! (b1)) -> stack! (b64) {
    var! {
        w : stack! (b64);
    }
    code! {
        w = x;
        w = add(w,x);
        if (w == 5) {
            (z, x) = add(x,w);
            (z, x) = add(x,y);
        }
    }
    return x
}

/*
START:CMD
ARG="renumber_reuse,print[input][types]"
END:CMD
*/
