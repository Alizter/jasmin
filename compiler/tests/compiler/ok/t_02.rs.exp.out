renumbering identifiers in all function(s)
   REMOVED_TIME
>> input:
#[macro_use] extern crate jasmin;

fn foo(mut x: stack! (b64), mut y: stack! (b64), mut z: reg! (b1)) -> stack! (b64) {
    var! {
        w: stack! (b64);
    }
    code! {
        w: stack! (b64) = x: stack! (b64);
        w: stack! (b64) = add(w: stack! (b64),x: stack! (b64));
        if (w: stack! (b64) == 5) {
            (z: reg! (b1),x: stack! (b64)) = add(x: stack! (b64),w: stack! (b64));
            (z: reg! (b1),x: stack! (b64)) = add(x: stack! (b64),y: stack! (b64));
        }
    }
    return x: stack! (b64)
}


total transformation time: REMOVED_TIME

total time (with save/print): REMOVED_TIME
